### ISSUE TITLE
how to use `System.Collections.Immutable` v`8.0.0` with `Microsoft.CodeAnalysis.Testing.ReferenceAssemblies.Net.Net60`

### ISSUE THREAD

**Issue by LukasGelke** [dotnet/roslyn-sdk#1155]
our code-analyzer-tests use `Microsoft.CodeAnalysis.CSharp.CodeFix.Testing.MSTest`. We now had to upgrade `System.Collections.Immutable`  to `8.0.0` for all Assemblies. Whenever we now want to use these assemblies to run within tests (because we have tests specifically for Classes inside our assemblies) we get an Exception:
```txt
// error CS1705: Assembly 'Komsa' with identity 'Komsa, Version=2.23.1.0, Culture=neutral, PublicKeyToken=null' uses 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version than referenced assembly 'System.Collections.Immutable' with identity 'System.Collections.Immutable, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
DiagnosticResult.CompilerError("CS1705").WithArguments("Komsa", "Komsa, Version=2.23.1.0, Culture=neutral, PublicKeyToken=null", "System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", "System.Collections.Immutable", "System.Collections.Immutable, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"),
```

adding
```cs
new Verifier.Test()
{
  TestState =
  {
    AdditionalReferences = { typeof(ImmutableArray).Assembly  },
  }
}
```
results in the following:
```txt
Actual diagnostic:
    // /0/Test0.cs(13,5): error CS0433: The type 'ImmutableArray<T>' exists in both 'System.Collections.Immutable, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' and 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
DiagnosticResult.CompilerError("CS0433").WithSpan(13, 5, 13, 27).WithArguments("System.Collections.Immutable, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", "System.Collections.Immutable.ImmutableArray<T>", "System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"),
```

adding the NugetPackage like this does not change anything (in either above case):
```cs
new Verifier.Test()
{
  ReferenceAssemblies = ReferenceAssemblies.Net.Net60 // used always
    .AddPackages([new PackageIdentity("System.Collections.Immutable", "8.0.0")]) // new
}
```

Our verifiers are essentially a "thin wrapper" around `CodeFixVerifier<TAnalyzer, TCodeFix, TTest, TVerifier>` and Verifier.Test around `CSharpCodeFixTest<TAnalyzer, TCodeFix, TVerifier>`, to allow for more testing and better migrating from our old tests, and as a workaround for other bugs in the past (ie tests not being able to handle 2-space indents correctly). We do not interfere with Roslyn in this subclass, we "only extend/append" (editorconfig-spacing) on what the base implementation provides. Thus I don't see us being the reason for "something not working"

**Comment by sharwell** [comment]
This is expected to work:

```
  ReferenceAssemblies = ReferenceAssemblies.Net.Net60 // used always
    .AddPackages([new PackageIdentity("System.Collections.Immutable", "8.0.0")]) // new
```

Someone would need to add a unit test to [`MetadataReferenceTests`](https://github.com/dotnet/roslyn-sdk/blob/68ac704c81fc9832a8120799fc6842b643f5a45e/tests/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing.UnitTests/MetadataReferenceTests.cs) and debug through why it's not working.

**Comment by onyxmaster** [comment]
Sam, thanks for fixing this (this bit us today with `S.C.Immutable 9.0.0` vs `ReferenceAssemblies.Net.Net80`).
Could you please tell us what's the release policy for the `Microsoft.CodeAnalysis.Analyzer.Testing` package? There has been no update since this fix, is there some kind of a schedule for the releases or is it blocked by something else?

**Comment by sharwell** [comment]
@onyxmaster I would expect this change is included in 1.1.3-beta1.24412.1 and newer. Packages are published to this feed after each pull request merge:
https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet-tools

**Comment by onyxmaster** [comment]
Sam, thank you very much, I'll take a look at it next week!
